### PR TITLE
Update VNC documentation to support Bookworm VNC stack

### LINKED ISSUE #2904: VNC changes
Presumably due to the switch from X11 to Wayland, VNC has changed from RealVNC to https://github.com/any1/wayvnc.

### PR DESCRIPTION
**Pull request by nathan-contino**
Closes #2904 

* Updated server-side instructions to start the WayVNC server using the new options in RPi Configuration
* Took the liberty of making the "find your IP address" documentation a little more straightforward for the average user (the previous version hid the easiest command in a NOTE!), and to add a new way of finding your IP address using `nmcli` that I discovered when I worked on the network configuration section
* Removed instructions for removed features:
  * RealVNC-specific cloud connections
  * experimental single-application VNC connections

And of course the usual sprucing up and streamlining of surrounding text so I can earn my "leave it better than you found it" Scout badge.

### CODE REVIEW

File: documentation/asciidoc/computers/remote-access/remote-access-introduction.adoc [hunk 1]
```diff
@@ -9,13 +9,91 @@ Any device connected to a Local Area Network is assigned an IP address. In order
...
+GENERAL.MTU:                            1500
+GENERAL.STATE:                          20 (unavailable)
+GENERAL.CONNECTION:                     --
+GENERAL.CON-PATH:                       --
+WIRED-PROPERTIES.CARRIER:               off
+IP4.GATEWAY:                            --
+IP6.GATEWAY:                            --
+----
+
+This command outputs information about the various network interfaces accessible on your Raspberry Pi. Check the `GENERAL.TYPE` row to see which kind of network interface each block describes. For example, "eth0" is the ethernet port on your device, and "wifi" refers to the WiFi chip built into some devices. You'll look at different blocks of output to find your IP address depending on the way your device accesses the internet:
```
Comment (lurch) [inline documentation/asciidoc/computers/remote-access/remote-access-introduction.adoc:89]: Small nitpick, but in the code-block above, the `GENERAL.TYPE` lists "ethernet", and "eth0" is the `GENERAL.DEVICE`.

File: documentation/asciidoc/computers/remote-access/remote-access-introduction.adoc [hunk 2]
```diff
@@ -9,13 +9,91 @@ Any device connected to a Local Area Network is assigned an IP address. In order
...
+In a web browser, navigate to your router's IP address. This is often `http://192.168.1.1`, but you may be able to find it printed on a label on your router. This will take you to a control panel. Then log in using your credentials, which is usually also printed on the router or sent to you in the accompanying paperwork. Browse to the list of connected devices or similar (all routers are different), and you should see some devices you recognise. Some devices are detected as PCs, tablets, phones, printers, etc. so you should recognise some and rule them out to figure out which is your Raspberry Pi. Also note the connection type; if your Raspberry Pi is connected with a wire there should be fewer devices to choose from.
+
+==== Network Manager
+
+You can use the built-in Network Manager CLI (`nmcli`) to access details about your network. Run the following command:
```
Comment (lurch) [inline documentation/asciidoc/computers/remote-access/remote-access-introduction.adoc:32]: Does this (fairly verbose) section offer anything extra that the earlier (much simpler) `hostname -I` command doesn't? :shrug: 

Might also be worth mentioning that you can mouse-hover over the network-icon in the systray, and the tooltip will display your IP address.

File: documentation/asciidoc/computers/remote-access/vnc.adoc [hunk 3]
```diff
@@ -1,123 +1,85 @@
...
 
-VNC Connect from RealVNC is included with Raspberry Pi OS. It consists of both VNC Server, which allows you to control your Raspberry Pi remotely, and VNC Viewer, which allows you to control desktop computers remotely from your Raspberry Pi should you want to.
+Raspberry Pi OS includes https://github.com/any1/wayvnc[wayvnc]. This provides a VNC server that you can enable in your device preferences.
 
-You must enable VNC Server before you can use it. By default, VNC Server gives you remote access to the graphical desktop that is running on your Raspberry Pi, as though you were sitting in front of it.
+Before you can use VNC on your Raspberry Pi, you must enable the VNC server.
 
-However, you can also use VNC Server to gain graphical remote access to your Raspberry Pi if it is headless or not running a graphical desktop. For more information on this, see *Creating a virtual desktop*, further below.
+NOTE: Previous versions of Raspberry Pi OS supported client connections from RealVNC. In version _Bookworm_ and above, Raspberry Pi OS no longer supports client connections from RealVNC. Instead, you can connect using TigerVNC.
```
Comment (lurch) [inline documentation/asciidoc/computers/remote-access/vnc.adoc:15]: I think we prefer "In Raspberry Pi OS _Bookworm_ or later, ..." rather than "In version _Bookworm_ and above, Raspberry Pi OS ..."

File: documentation/asciidoc/computers/remote-access/vnc.adoc [hunk 4]
```diff
@@ -1,123 +1,85 @@
...
 
-=== Installing VNC on Raspberry Pi
+=== Install VNC on Raspberry Pi
 
-VNC is already installed on the full Raspberry Pi OS image, and can be installed via `Recommended Software` from the `Preferences` menu on other versions.
+Raspberry Pi OS comes with WayVNC already installed. If you run a version of Raspberry Pi OS that does not include WayVNC, you can:
```
Comment (lurch) [inline documentation/asciidoc/computers/remote-access/vnc.adoc:19]: If Raspberry Pi OS comes with WayVNC already installed, I don't think that we need to provide instructions on how to install it?

**Comment by lurch** [comment]
IIRC only Pi 4 is running the Wayland backend (and therefore uses WayVNC); whereas on earlier Pis the X11 backend is still used, and enabling "VNC server" in `raspi-config` or the preferences GUI still runs the old RealVNC server? ping @spl237

**Comment by spl237** [comment]
> IIRC only Pi 4 is running the Wayland backend (and therefore uses WayVNC); whereas on earlier Pis the X11 backend is still used, and enabling "VNC server" in `raspi-config` or the preferences GUI still runs the old RealVNC server? ping @spl237

Correct.

File: documentation/asciidoc/computers/remote-access/vnc.adoc (hunk 4 above)
Comment (aallan) [reply documentation/asciidoc/computers/remote-access/vnc.adoc:19]: Yup. The installation steps were a holdover from the days when it didn't. The VNC server comes pre-installed on both desktop and lite distros, don't need to step through installation.

**Comment by aallan** [comment]
> > IIRC only Pi 4 is running the Wayland backend (and therefore uses WayVNC); whereas on earlier Pis the X11 backend is still used, and enabling "VNC server" in `raspi-config` or the preferences GUI still runs the old RealVNC server? ping @spl237
> 
> Correct.

Oh, that's actually a pain in the backside. Fairly sure TigerVNC isn't going to connect to Real's server running on pre-Pi4 hardware. Which means that we're going to have to point to different client side tools depending on model.

@spl237 do you know if the TigerVNC client will connect to Real VNC's server on still-running-X11 Pis?

**Comment by spl237** [comment]
> > > IIRC only Pi 4 is running the Wayland backend (and therefore uses WayVNC); whereas on earlier Pis the X11 backend is still used, and enabling "VNC server" in `raspi-config` or the preferences GUI still runs the old RealVNC server? ping @spl237
> > 
> > 
> > Correct.
> 
> Oh, that's actually a pain in the backside. Fairly sure TigerVNC isn't going to connect to Real's server running on pre-Pi4 hardware. Which means that we're going to have to point to different client side tools depending on model.
> 
> @spl237 do you know if the TigerVNC client will connect to Real VNC's server on still-running-X11 Pis?

No idea - I can't see any reason why it shouldn't, but I haven't tried it.

**Comment by aallan** [comment]
RealVNC uses proprietary encryption, not implemented by other clients. I’m not sure their server offers something that not-their-client will connect to.

Needs testing with a not-a-Pi4 running Bookworm, and Real’s client on someone’s laptop. I’m not back in the office on Tuesday.

File: documentation/asciidoc/computers/remote-access/remote-access-introduction.adoc (hunk 2 above)
Comment (nathan-contino) [reply documentation/asciidoc/computers/remote-access/remote-access-introduction.adoc:32]: Good point. This section probably *could* be eliminated, but I also like the fact that if you have multiple active network interfaces -- say, you're plugged into ethernet *and* you're connected to wifi -- this will show you IPs for both. The latter sections are arguably even more esoteric, so... I would argue that if we want to eliminate this for redundancy, the same logic applies to the next few subsections.

FANTASTIC point about the mouse-hover. I did not know that, and honestly that should be the first suggestion in this whole section, since it doesn't require any CLI knowledge at all! Adding it now.

File: documentation/asciidoc/computers/remote-access/vnc.adoc (hunk 3 above)
Comment (nathan-contino) [reply documentation/asciidoc/computers/remote-access/vnc.adoc:15]: Maybe you can help me out here -- I originally used that wording, but I ended up with *three* Proper Noun Namings of Raspberry Pi OS in the space of two sentences. So I eliminated the only optional one to keep sentences in the active voice. I'll see if I can reword it to keep the preferred Bookworm name convention but if you have suggestions my ears are open.

File: documentation/asciidoc/computers/remote-access/vnc.adoc (hunk 4 above)
Comment (nathan-contino) [reply documentation/asciidoc/computers/remote-access/vnc.adoc:19]: The old docs included install information, so I preserved it here. Apparently the Lite distribution didn't include the old VNC stack; I haven't had enough SD cards to verify if Bookworm includes VNC stuff in its lite distro (but I'm assuming it doesn't, since it doesn't seem essential).

**Comment by nathan-contino** [comment]
I have a Pi running an earlier OS laying around. I'll try enabling VNC and see if I can connect with TigerVNC. As long as that works, I think it's simpler and easier to stick with TigerVNC instructions for all users, instead of bloating the docs with two separate third-party tool instruction sets.

File: documentation/asciidoc/computers/remote-access/vnc.adoc (hunk 4 above)
Comment (nathan-contino) [reply documentation/asciidoc/computers/remote-access/vnc.adoc:19]: Scratch that; GitHub didn't show me Al's comment earlier. I'll remove the installation instructions then! Fewer words better.

File: documentation/asciidoc/computers/remote-access/find-your-ip-address.adoc [hunk 5]
```diff
@@ -0,0 +1,225 @@
+[[ip-address]]
+=== How to Find your IP Address
+
+To find the local IP of your Raspberry Pi, use one of the following methods.
```
Comment (lurch) [inline documentation/asciidoc/computers/remote-access/find-your-ip-address.adoc:4]: Probably want to use "IP address" rather than just "IP" ?

File: documentation/asciidoc/computers/remote-access/find-your-ip-address.adoc [hunk 6]
```diff
@@ -0,0 +1,225 @@
...
+
+If you use a display with your Raspberry Pi and you boot to the command line instead of the desktop, the boot sequence will include your IP address as one of the last few output messages before your login prompt.
+
+==== Router devices list
+
+In a web browser, navigate to your router's IP address. This is often `http://192.168.1.1`, but you may be able to find it printed on a label on your router. This will take you to a control panel. Then log in using your credentials, which is usually also printed on the router or sent to you in the accompanying paperwork. Browse to the list of connected devices or similar (all routers are different), and you should see some devices you recognise. Some devices are detected as PCs, tablets, phones, printers, etc. so you should recognise some and rule them out to figure out which is your Raspberry Pi. Also note the connection type; if your Raspberry Pi is connected with a wire there should be fewer devices to choose from.
+
+==== Network Manager
+
+You can use the built-in Network Manager CLI (`nmcli`) to access details about your network. Run the following command:
```
Comment (lurch) [inline documentation/asciidoc/computers/remote-access/find-your-ip-address.adoc:30]: I wonder if this should be coalesced into the "Command Line" section? Perhaps something like "If you want further information, you can also use Network Manager CLI (`nmcli`) ..." ? :shrug:

File: documentation/asciidoc/computers/remote-access/find-your-ip-address.adoc [hunk 7]
```diff
@@ -0,0 +1,225 @@
...
+
+----
+hostname -I
+----
+
+==== Boot Output
+
+If you use a display with your Raspberry Pi and you boot to the command line instead of the desktop, the boot sequence will include your IP address as one of the last few output messages before your login prompt.
+
+==== Router devices list
```
Comment (lurch) [inline documentation/asciidoc/computers/remote-access/find-your-ip-address.adoc:24]: Might be worth moving this to below the raspberrypi.local section? (so that the various approaches are ordered from easiest -> most difficult)

File: documentation/asciidoc/computers/remote-access/find-your-ip-address.adoc [hunk 8]
```diff
@@ -0,0 +1,225 @@
...
+
+In a web browser, navigate to your router's IP address. This is often `http://192.168.1.1`, but you may be able to find it printed on a label on your router. This will take you to a control panel. Then log in using your credentials, which is usually also printed on the router or sent to you in the accompanying paperwork. Browse to the list of connected devices or similar (all routers are different), and you should see some devices you recognise. Some devices are detected as PCs, tablets, phones, printers, etc. so you should recognise some and rule them out to figure out which is your Raspberry Pi. Also note the connection type; if your Raspberry Pi is connected with a wire there should be fewer devices to choose from.
```
Comment (lurch) [inline documentation/asciidoc/computers/remote-access/find-your-ip-address.adoc:26]: This is often `http://192.168.0.1` or `http://192.168.1.1`, but ...

File: documentation/asciidoc/computers/remote-access/find-your-ip-address.adoc [hunk 9]
```diff
@@ -0,0 +1,225 @@
...
+GENERAL.CON-PATH:                       --
+WIRED-PROPERTIES.CARRIER:               off
+IP4.GATEWAY:                            --
+IP6.GATEWAY:                            --
+----
+
+This command outputs information about the various network interfaces accessible on your Raspberry Pi. Check the `GENERAL.TYPE` row to see which kind of network interface each block describes. For example, "ethernet" is the ethernet port on your device, and "wifi" refers to the WiFi chip built into some devices. You'll look at different blocks of output to find your IP address depending on the way your device accesses the internet:
+
+* if your device connects to the internet using WiFi, check the "wifi" block
+* if your device connects to the internet using the ethernet port, check the "eth0" block
```
Comment (lurch) [inline documentation/asciidoc/computers/remote-access/find-your-ip-address.adoc:90]: Another eth0 -> ethernet change needed here.

File: documentation/asciidoc/computers/remote-access/find-your-ip-address.adoc [hunk 10]
```diff
@@ -0,0 +1,225 @@
...
+----
+
+If the Raspberry Pi is reachable, `ping` will show its IP address:
+
+----
+PING raspberrypi.local (192.168.1.131): 56 data bytes
+64 bytes from 192.168.1.131: icmp_seq=0 ttl=255 time=2.618 ms
+----
+
+If you change the system hostname of the Raspberry Pi (e.g., by editing `/etc/hostname`), Avahi will also change the `.local` mDNS address.
```
Comment (lurch) [inline documentation/asciidoc/computers/remote-access/find-your-ip-address.adoc:115]: Should also mention here about the hostname being configurable via Raspberry Pi Imager.

File: documentation/asciidoc/computers/remote-access/find-your-ip-address.adoc [hunk 11]
```diff
@@ -0,0 +1,225 @@
...
+
+`fd00::ba27:ebff:feb6:f293 2001:db8:494:9d01:ba27:ebff:feb6:f293`
+
+The example shows two IP addresses. The first one is a so called unique local unicast address(`fc00::/7`). The second one is the global unicast address(`2000::/3`). It is also possible to see only one of them depending on your network (router) configuration. Both addresses are valid for reaching the Raspberry Pi within your LAN. The address out of `2000::/3` is accessible world wide, provided your router's firewall is opened.
+
+Now use one of IPs from the first step to ping all local nodes:
+
+----
+ping -c 2 -I 2001:db8:494:9d01:ba27:ebff:feb6:f293  ff02::1
+ping -c 2 -I 2001:db8:494:9d01:ba27:ebff:feb6:f293  ff02::1%eth0
```
Comment (lurch) [inline documentation/asciidoc/computers/remote-access/find-your-ip-address.adoc:171]: Why are there two ping commands here?

File: documentation/asciidoc/computers/remote-access/find-your-ip-address.adoc [hunk 12]
```diff
@@ -0,0 +1,225 @@
...
+64 bytes from raspberrypi.fritz.box (2001:db8:494:9d01:ba27:ebff:feb6:f293): icmp_seq=2 ttl=64 time=0.480 ms
+
+--- ip6-allnodes ping statistics ---
+2 packets transmitted, 2 received, +5 duplicates, 0% packet loss, time 1001ms
+rtt min/avg/max/mdev = 0.480/1.283/2.623/0.735 ms
+----
+
+This should result in replies from all the nodes on your (W)LAN link, with associated DNS names.
+
+Exclude your own IP( here `2001:db8:494:9d01:ba27:ebff:feb6:f293` ),
```
Comment (lurch) [inline documentation/asciidoc/computers/remote-access/find-your-ip-address.adoc:202]: Weird spacing around your brackets.

File: documentation/asciidoc/computers/remote-access/find-your-ip-address.adoc (hunk 11 above)
Comment (nathan-contino) [reply documentation/asciidoc/computers/remote-access/find-your-ip-address.adoc:171]: Honestly I'm not sure -- I left the command as-is because I don't fully understand it and I'm trying to limit scope creep :)

File: documentation/asciidoc/computers/remote-access/find-your-ip-address.adoc [hunk 13]
```diff
@@ -0,0 +1,225 @@
...
+Here you can see a device with hostname `raspberrypi` has IP address `192.168.1.8`. Note, to see the hostnames, you must run nmap as root by prepending `sudo` to the command.
+
+==== Getting IPv6 addresses by pinging from a second device 
+
+First find your own IP address(es), in other words the one of the computer you're using to find your Raspberry Pi's IP address
+by `hostname -I`
+
+`fd00::ba27:ebff:feb6:f293 2001:db8:494:9d01:ba27:ebff:feb6:f293`
+
+The example shows two IP addresses. The first one is a so called unique local unicast address(`fc00::/7`). The second one is the global unicast address(`2000::/3`). It is also possible to see only one of them depending on your network (router) configuration. Both addresses are valid for reaching the Raspberry Pi within your LAN. The address out of `2000::/3` is accessible world wide, provided your router's firewall is opened.
```
[the remark this answers is not in the document]
Comment (nathan-contino) [reply documentation/asciidoc/computers/remote-access/find-your-ip-address.adoc]: I'm not sure we need to clarify that in this paragraph -- it's already a meaty and confusing subject, and I fear we'll lose a good chunk of our remaining readers on this if we add more meat to the pile.

Fixed the spaces though 😄

File: documentation/asciidoc/computers/remote-access/find-your-ip-address.adoc (hunk 10 above)
Comment (nathan-contino) [reply documentation/asciidoc/computers/remote-access/find-your-ip-address.adoc:111]: The next paragraph mentions that you can change the hostname, that's good enough for me.

File: documentation/asciidoc/computers/remote-access/find-your-ip-address.adoc (hunk 7 above)
Comment (nathan-contino) [reply documentation/asciidoc/computers/remote-access/find-your-ip-address.adoc]: Agreed, moved.

File: documentation/asciidoc/computers/remote-access/find-your-ip-address.adoc (hunk 6 above)
Comment (nathan-contino) [reply documentation/asciidoc/computers/remote-access/find-your-ip-address.adoc:26]: Debatable. Many of the other sections are *also* command line... just in increasingly complicated ways.

Personally I'm in favor of eliminating all of the sections other than "command line", "desktop", and maybe one alternative for advanced use cases.

Comment (nathan-contino) [reply documentation/asciidoc/computers/remote-access/find-your-ip-address.adoc:26]: But I'm also willing to table that discussion out of respect to Scope Creep.

Comment (lurch) [reply documentation/asciidoc/computers/remote-access/find-your-ip-address.adoc:26]: Probably a decision for @aallan then :slightly_smiling_face:

File: documentation/asciidoc/computers/remote-access/find-your-ip-address.adoc (hunk 10 above)
Comment (lurch) [reply documentation/asciidoc/computers/remote-access/find-your-ip-address.adoc:111]: With the difference being that Raspberry Pi Imager lets you configure the hostname _before_ you even flash the image to SD card, so no need to muck about with trying to change it after you've booted up the OS (which may be tricky, if you're running headless).

File: documentation/asciidoc/computers/remote-access/find-your-ip-address.adoc (hunk 13 above)
[the remark this answers is not in the document]
Comment (lurch) [reply documentation/asciidoc/computers/remote-access/find-your-ip-address.adoc]: Fair enough about the meaty IPv6 pile :laughing: 
There's still one of the mising-spaces that you've missed though :wink:

File: documentation/asciidoc/computers/remote-access/find-your-ip-address.adoc (hunk 11 above)
Comment (lurch) [reply documentation/asciidoc/computers/remote-access/find-your-ip-address.adoc:171]: Right; it was only towards the end of reviewing this that I realised that you were just moving existing content from one section to another, and not adding new content here :facepalm:

**Comment by lurch** [comment]
> Needs testing with a not-a-Pi4 running Bookworm, and Real’s client on someone’s laptop.

Or a Pi 4 running Bookworm, where you've used `raspi-config` to switch it to using the X11 backend, before enabling VNC server.

File: documentation/asciidoc/computers/remote-access/find-your-ip-address.adoc [hunk 14]
```diff
@@ -0,0 +1,225 @@
+[[ip-address]]
+=== How to Find your IP Address
```
[the remark this answers is not in the document]
Comment (aallan) [reply documentation/asciidoc/computers/remote-access/find-your-ip-address.adoc]: Plus I like Title Case, so a lot of the new headings are Title Case, at least until Liz notices and then they aren't.

**Comment by nathan-contino** [comment]
Can confirm that switching a Pi 4 to the X11 backend and enabling the VNC server still works.
Not-a-Pi running Bookworm also works.

**Comment by aallan** [comment]
> Can confirm that switching a Pi 4 to the X11 backend and enabling the VNC server still works.
> 
> Not-a-Pi running Bookworm also works.

Right. Document TigerVNC as the client at that point. If we don't have to document two clients. That's a win and reduces complexity a bunch. 

Could probably do with some client side screenshots.

**Comment by nathan-contino** [comment]
Added hopefully-not-too-many client and server graphical setup screenshots! And made some (minor) improvements to the IPv6 wording. Though the IPv6 section is clunky enough that I'm tempted to fully delete it, since the new `nmcli` instructions *should* show IPv6 addresses as well, if you're one of the lucky few living in the idealistic future with enough addresses for everyone.

**Comment by aallan** [comment]
> Though the IPv6 section is clunky enough that I'm tempted to fully delete it, since the new nmcli instructions should show IPv6 addresses as well, if you're one of the lucky few living in the idealistic future with enough addresses for everyone.

Kill IPv6 with fire 🔥 

The user base going to these doc's ain't using it, and folks that are, don't need these docs.

**Comment by aallan** [comment]
After you pull the IPv6 stuff I think we should merge.

**Comment by aallan** [comment]
Ready to merge?

**Comment by nathan-contino** [comment]
Ready to merge! @aallan

**Review by aallan — APPROVED** [review]
YOLO
